### PR TITLE
 Static content support for razor class libraries

### LINKED ISSUE #12486: [Preview 6] Static content support for razor class libraries

### PR DESCRIPTION
**Pull request by javiercn**
[Fixes #12486]

### CODE REVIEW

**Comment by guardrex** [comment]
Why isn't this being added to the existing topic that pertains to RCL?

https://docs.microsoft.com/aspnet/core/blazor/class-libraries?view=aspnetcore-3.0&tabs=visual-studio

**Comment by javiercn** [comment]
https://github.com/aspnet/AspNetCore.Docs/pull/12695/files?short_path=49e2a14#diff-49e2a14c4922e0e0c05459e818dcb7d2

**Comment by Rick-Anderson** [comment]
@javiercn  is it OK if I move this to [Blazor Razor components class libraries](https://docs.microsoft.com/en-us/aspnet/core/blazor/class-libraries?view=aspnetcore-3.0&tabs=visual-studio)

**Comment by javiercn** [comment]
Go for it

**Comment by danroth27** [comment]
@Rick-Anderson The new static content support in razor class libraries applies not only to Blazor, but also when using Razor Pages and MVC. Can we have a cross-cutting Razor Class Libraries topic under the "Web apps" section?

**Comment by Rick-Anderson** [comment]
@danroth27  I could put all that content in an include file and include it in both topics. WOuld that work?

**Comment by danroth27** [comment]
> I could put all that content in an include file and include it in both topics. WOuld that work?

@Rick-Anderson I don't think that's really necessary. What I would suggest is to put this content in the existing [Razor Class Library](https://docs.microsoft.com/en-us/aspnet/core/razor-pages/ui-class) topic. We should then move that topic to be a top level topic in the "Web Apps" node instead of being under Razor Pages since Razor class libraries are not specific to Razor Pages. We then add a brief section in the "Razor Class Library" doc that says that Razor class libraries can also be used to build component class libraries, and then link to the existing [component class libraries topic](https://docs.microsoft.com/en-us/aspnet/core/blazor/class-libraries) under the Blazor node where we cover the details of creating libraries of components and consuming them. How does that sound?

File: aspnetcore/blazor/multi-project content support.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+monikerRange: '>= aspnetcore-3.0'
+ms.author: riande
+ms.custom: mvc
+ms.date: 06/26/2019
+uid: blazor/multi-project-content-support.md
+---
+
+# Create a razor class library with static assets
+
+Razor Class Libaries (RCL) frequently require companion static assets that can be to be referenced by the consuming app of the RCL. ASP.NET Core allows creating RCLs that include static assets that are available from a consuming app.
```
Comment (Andrzej-W) [inline aspnetcore/blazor/multi-project content support.md]: > that can be to be referenced

This have to be corrected

Comment (Rick-Anderson) [reply aspnetcore/blazor/multi-project content support.md]: @Andrzej-W I'll take care of the editing.

**Comment by Rick-Anderson** [comment]
@tdykstra I'm closing this PR. Create a new one and apply changes in a new PR per `@danroth27` instructions.

**Comment by guardrex** [comment]
I think this has already been done. Sorry I didn't cross-link. I'll track it down and cross-link it here.

**Comment by tdykstra** [comment]
@guardrex - Doesn't look like it's been done yet:
* RCL doc is still under Razor Pages in the TOC
* RCL doc doesn't have the new content from this PR
* RCL doc still needs section that says Razor class libraries can be used to build component class libraries, links to component class libraries doc

I'll start a PR to make those changes, unless you have them waiting already in an unmerged PR.

The component class libraries doc currently says:  “In ASP.NET Core 3.0 Preview, Blazor server-side apps can't consume static assets from … a Razor class library” – is that now incorrect?

**Comment by guardrex** [comment]
It's on (some of it anyway) https://github.com/aspnet/AspNetCore.Docs/pull/12911.

I think whatever we do should go on my existing PR.

**Comment by guardrex** [comment]
We had a few PRs that brought the content in and then modified it.

That one ... #12911 ... might not have everything given the scattered issues.

I'll have a break the current issue that I'm working shortly ... I'll go over that PR and make sure that the items you listed from this issue are addressed.

**Comment by tdykstra** [comment]
OK.  Looks like that also means #12911 shouldn't create the includes it currently creates.  And it should pick up the latest version of the content from this PR if it has an earlier version.

**Comment by guardrex** [comment]
I think the content is later on my PR, but I see what you mean about the INCLUDES because we'd cross-link the Blazor topic to the RCL topic for that coverage on static assets. IIRC, there was a reason why it doesn't do it, but I need to get back and look. Let me take a closer look when I get free shortly. I'm 🏃😅 at the moment on another issue.

**Pull request closed without merging**